json/src/main/java/org/apache/james/json/JsonGenericSerializer.java

```diff
@@ -107,10 +107,20 @@ public T deserialize(String value) throws IOException {
         }
     }
 
+    public T deserialize(String type, String value) throws IOException {
+        try {
+            JsonNode jsonNode = objectMapper.readTree(value);
```

Comment:
what's the point of reading it to a tree then read it to a value?

Reply:
We need to read the type, in order to know which DTO class in which we want to deserialize it.

Reply:
It's not what the code says. The type is a parameter and you don't use `jsonNode` but in readValue